python/tests/test_base_install/test_filters/test_exploded_edge_filter.py

```diff
@@ -0,0 +1,898 @@
+from raphtory import Graph, PersistentGraph
+from raphtory import filter
+import pytest
+from datetime import datetime
+
+
+@pytest.mark.parametrize("GraphClass", [Graph, PersistentGraph])
+def test_graph(GraphClass):
+
+    g = GraphClass()
+    g.add_edge(1, 1, 2, layer="blue", properties={"weight": 1, "name": "bob"})
+    g.add_edge(2, 1, 2, layer="blue", properties={"weight": 2, "name": "dave"})
+    g.add_edge(3, 1, 2, layer="blue", properties={"weight": 3, "name": "greg"})
+
+    g.add_edge(1, 1, 3, layer="blue", properties={"weight": 1, "name": "bob"})
+    g.add_edge(2, 1, 3, layer="blue", properties={"weight": 2, "name": "dave"})
+    g.add_edge(3, 1, 3, layer="red", properties={"weight": 3, "name": "greg"})
+
+    weight_e3 = filter.Property("weight") == 3
+    weight_lt3 = filter.Property("weight") < 3
+    name_bob = filter.Property("name") == "bob"
+
+    f_g = g.filter_exploded_edges(filter=weight_e3)
+    e1 = f_g.edge(1, 2)
+    e2 = f_g.edge(1, 3)
+
+    if type(g) == Graph:
+        assert e1.deletions() == []
+        assert e2.deletions() == []
+    else:
+        assert e1.deletions() == [1, 2]
+        assert e2.deletions() == [1, 2]
+
+    assert list(e1.history()) == [3]
+    assert list(e2.history()) == [3]
+
+    # assert e2.layer_names == ["red"] returning red blue for PersistentGraph which feels wrong?
```

Comment:
that is consistent with how we are handling edges that are only deleted right now (so probably ok of this version).